sys/shell/commands/sc_gnrc_netif.c

```diff
@@ -31,6 +31,9 @@
 #ifdef MODULE_L2FILTER
 #include "net/l2filter.h"
 #endif
+#ifdef MODULE_SX127X
+#include "net/lora.h"
+#endif
```

Comment:
Please remove all those `#ifdef`s and `#ifndef`s. They are not necessary. Especially the `#ifndefs` shouldn't be there. And if you want to keep the `#ifdef`s, please use another module (e.g. [a pseudo-module] `netdev_lora`).

Reply:
I know I can remove them but if I do that I have 2 problems:
* unrelated options/parameters are displayed in the usage output
* the size of firmware increases for other boards (e.g samr21-xpro)

I'll change that for the pseudo module.

Reply:
> unrelated options/parameters are displayed in the usage output

a) That wasn't a problem for other types of devices. A user who has only an Ethernet device also sees the options to set a channel, but will notice it was stupid to try to set them as soon as they see the error message `ifconfig` reports, because the device does not support them.
b) You exclude the IPv6-options. IPv6 over LoRA is [in the works](https://datatracker.ietf.org/wg/lpwan/about/), so removing them seems a little short-sighted.
c) If a board has two interfaces of which only one board is a LoRa board, the user wouldn't be able to configure the other interface properly, because the LoRA device is compiled in. This is a valid scenario, since e.g. LoRA-to-IEEE802.15.4 border routers is something we already sketched out.

> the size of firmware increases for other boards (e.g samr21-xpro)

That is a valid argument, but only if you make it dependent on LoRA, not on the device.

Reply:
Added the pseudo module and removed a few ifdefs.